File: Ywy/JavaScript/level1/DivisibleNumber.js
// const solution = (arr, divisor) => {
//     let answer = [];
//     let temp = 0;
//     arr.forEach( el => {
//         temp = el / divisor;
//         if (/^[1-9]{1}[0-9]*$/.test(temp) || /^[1-9]{1}[0-9]*[.]{1}[0]{1}$/.test(temp)) {
//             answer.push(el);
//         }
//     });
//     answer = answer.sort((prev, next) => prev - next);
//     return answer[0] === undefined ? [-1] : answer;
// };

const solution = (arr, divisor) => {
    const answer = [];
    arr.forEach( el => {
        if (el % divisor === 0) {
            answer.push(el);
        }
    });
    return answer[0] === undefined ? [-1] : answer.sort((prev, next) => prev - next);
};

File: Ywy/JAVA/src/level1/DivisibleNumber.java
package level1;

public class DivisibleNumber {

	public int[] solution(int[] arr, int divisor) {
        int index = 0;
        int range = arr.length;
        int addIndex = 0;
        int element = 0;
        double divi = divisor * 1.0;
        do {
            element = arr[index];
            if (check(element / divi)) {
                arr[addIndex++] = element;
            }
        } while(++index < range);
        if (addIndex == 0) {
            return new int[]{-1};
        }
        final int[] answer = new int[addIndex];
        System.arraycopy(arr, 0, answer, 0, addIndex);
        java.util.Arrays.sort(answer);
        return answer;
    }
    
    private boolean check(double d) {
        String temp = String.valueOf(d);
        int dotIndex = temp.indexOf(".");
        if (temp == null || temp.equals("0") || dotIndex == -1) {
            return false;
        }
        temp = temp.replace(".", "");
        return strToInt(temp.substring(0, dotIndex)) > 0 &&
               temp.substring(dotIndex, temp.length()).equals("0");
    }
    
    private long strToInt(String str) {
        return Long.parseLong(str);
    }

}
